nujo/utils/data/dataset_loader.py

```diff
@@ -12,34 +13,77 @@ class DatasetLoader:
 
     Parameters:
     -----------
-    name : will be downloaded from the UCI ML repo
+    - name : str,
+    - type : str, indicates the type of file, can be csv, image or mnist
+    - override : bool, if this file exists, does it get downloaded again
     '''
-    _UCI_REPO_URL = 'archive.ics.uci.edu/ml/machine-learning-databases/{}/{}'
+    _UCI_REPO_URL = (
+        'https://archive.ics.uci.edu/ml/machine-learning-databases/{}/{}')
 
-    def __init__(self, dataset):
-        self.name = dataset.name  # with .data
-        self._link = self._UCI_REPO_URL.format(self.name[:-4], self.name)
-        with open(self.name, 'r+') as data:
-            lines = data.readlines()
+    def __init__(self, name: str, type: str, override: bool = True):
+        self.name = name.strip().lower()
+        self.type = type.strip().lower()
+        self._filepath = f'{HOME_DIR}{self.name}.data'
+        if not override and exists(self._filepath):
+            return
 
-        dataset.X = empty((0, len(lines[0].split(','))))
-        # number of columns
-        for line in lines[:-1]:  # last row is \n
-            x = array(line.strip().split(','))
-            dataset.X = vstack((dataset.X, x))
+    def install(self, filepath: str = None, labels: ndarray = None) -> ndarray:
```

Comment:
When a variable can be either of some type or `None` use `Optional[the_type]`
In this case use `Optional[str]` and `Optional[ndarray]`
`Optional` should be imported again from `typing`